app/models/work-item.ts

```diff
@@ -602,4 +617,44 @@ export async function getTotalWorkItemSizeForJobID(
   }
 
   return totalSize;
+}
+
+/**
+ * Compute the threshold (in milliseconds) to be used to expire work items for a given job/service
+ * 
+ * @param jobID - the ID of the Job for the step
+ * @param serviceID - the serviceID of the step within the workflow
+ */
+export async function computeWorkItemDurationOutlierThresholdForJobService(
+  jobID: string,
+  serviceID: string,
+): Promise<number> {
+  // default to two hours
+  let threshold = 7200000;
+
+  try {
+    // use a simple heuristic of 2 times the longest duration of all the successful work items
+    // for this job/service
+    const result = await db(WorkItem.table)
+      .where({
+        jobID,
+        serviceID,
+        'status': 'successful',
+      })
+      .max('duration', { as: 'max' })
+      .first();
+
+    if (result) {
+      threshold = 2.0 * result.max;
+    } else {
+      logger.debug('Using default threshold');
+    }
+    logger.debug(`Threshold is ${threshold}`);
+
+  } catch (e) {
+    logger.error(`Failed to get MAX duration for service ${serviceID} of job ${jobID}`);
+  }
+
+  return threshold;
```

Comment:
Do we need any sort of sanity check that this is a positive number since the default value for the duration column is negative?

Reply:
Done.